contracts/allowlist/ValidatorSmartContractAllowList.sol

```diff
@@ -0,0 +1,127 @@
+// Implementation of a contract to select validators using an allowlist
+
+pragma solidity >=0.5.0;
+
+import "../ValidatorSmartContractInterface.sol";
+
+contract ValidatorSmartContractAllowList is ValidatorSmartContractInterface {
```

Comment:
One thing missing here is events: As far as I see every state changing function we have should emit an event when called. Most importantly the ones we need to track ongoing elections.